tsconfig.app.json

```diff
@@ -3,7 +3,7 @@
   "extends": "./tsconfig.json",
   "compilerOptions": {
     "outDir": "./out-tsc/app",
-    "types": []
+    "types": ["node", "webpack-env"]
```

Comment:
sao chỗ này phải thêm types dạ?, bạn thử bỏ nguyên dòng này xem chắc là chạy á.

Reply:
Nếu không có nó sẽ không tìm được “module”. Nên cần install type definitions for node. Nó cho phép Webpack step above. Anh thử bên anh là hiểu và lỗi nha.

Reply:
@hoangvanhoa1701 a thử ùi nha, mình hog cần cái `webpack-env` đâu, chỗ `module['hot']` bạn đổi thành (module as any)['hot'] là dc, đằng nào thì chỗ hmrBoostrap cũng define `module: any` chỗ function 😆. đỡ thêm 1 dependency.

Reply:
với lại để type node 12 có thấp hem, hình như lts đang là 14 ùi mà ta 😄

Reply:
Dạ cái node 12 là sẵn rùi đó anh. Em tính lên 13.x mà sợ có vấn đề nên không lên theo hmr

Reply:
@sandangel  vậy cái node có nâng lên bản mới nhất không hay sao anh. Anh chốt xong mai em xoá cái webpack env và update node luôn rùi commit phát cuối.

Reply:
mình nghĩ để vậy cũng dc, chỉ là type thui. khi nào cần thì nâng. 😘

Reply:
<img width="852" alt="Screen Shot 2021-09-09 at 9 35 01 AM" src="https://user-images.githubusercontent.com/45493195/132612930-db5068d6-4058-44c3-8801-ffd0f0ef1310.png">
Dạ @sandangel  bên anh có bị vậy không anh. Hay do trình soạn code của bên em bị thôi ta?

Reply:
do cái `node` type nó nằm trong tsconfig.app.json nên bên kia ko nhận diện dc, nhưng mà ko lỗi đâu.